lib/tty/prompt/multi_list.rb

```diff
@@ -60,6 +60,7 @@ def keyspace(*)
         else
           return if @max && @selected.size >= @max
           @selected << active_choice
+          @selected.sort_by! { |choice| @choices.index(choice) }
```

Comment:
When you mentioned another data structure to keep the items ordered, I thought about having some choice index cache. So when a choice is added to internal `@choices` collection we store name and index pair in a reference table? That should speed up lookup/sorting. Though I don't know how efficient `index` call is itself?

Reply:
Given that maximum amount of options reasonable for a human processing is dozens, may be hundreds or indeed thousands (given such choice can be done only via filters) and that reverse index search is O(n) it will anyway finish within a blink of an eye, so I wonder which problem are we trying to solve by speeding up lookup/sorting?

Reply:
I had people using it with few thousands of options and had to introduce filter cache. The impact can be quickly seen when rendering. Want to make sure we consider if that's the best approach. The index search and sorting worst-case scenario can be O(n^2). I'm not advocating premature optimization for the sake of it but equally, I don't want to degrade the speed. I will do some benchmarking to see what the impact is as I'm curious. This is not a show stopper by any means.

Reply:
Right, thanks for explaining Piotr, I asked that precisely to understand which benchmark [size] to write. I'll experiment with the cache and let's test our benchmark against it, will update you soon on it

Reply:
@piotrmurach you were right, when I tried a multi_select of 10_000 choices, a simple select/deselect of the same item 1000 times took 40 seconds(!).

Good news is that I sped it up by 20x (see next commit):

without optimisation:
` 40.132000   0.032000  40.164000 ( 40.165195)`

with optimisation:
`1.872000   0.000000   1.872000 (  1.871306)`